app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.kt

```diff
@@ -55,25 +59,23 @@ class AddEditTaskViewModel @Inject constructor(
 
     private val taskId: String? = savedStateHandle[TodoDestinationsArgs.TASK_ID_ARG]
 
-    // A MutableStateFlow needs to be created in this ViewModel. The source of truth of the current
-    // editable Task is the ViewModel, we need to mutate the UI state directly in methods such as
-    // `updateTitle` or `updateDescription`
-    private val _uiState = MutableStateFlow(AddEditTaskUiState())
-    val uiState: StateFlow<AddEditTaskUiState> = _uiState.asStateFlow()
+    private val eventStateChanges = MutableSharedFlow<StateChange<AddEditTaskUiState>>()
 
-    init {
```

Comment:
Being able to lazily initiate loading with a flow was a nice side effect of use of `produceState`